tiledb/sm/query/sparse_index_reader_base.cc

```diff
@@ -350,15 +428,35 @@ Status SparseIndexReaderBase::compute_tile_bitmaps(
     max_range_size = std::max(max_range_size, range_num);
   }
 
-  // Process all tiles in parallel.
-  auto result_tiles_it = ResultTileIt(result_tiles);
-  auto status = parallel_for(
-      storage_manager_->compute_tp(), 0, result_tiles_it.size(), [&](uint64_t) {
-        // Take a result tile.
-        auto rt = result_tiles_it.get_next();
+  // Process all tiles/cell ranges in parallel.
+  auto status = parallel_for_2d(
+      storage_manager_->compute_tp(),
+      0,
+      result_tiles->size(),
+      0,
+      num_cell_ranges,
+      [&](uint64_t t, uint64_t cell_range_idx) {
```

Comment:
I would avoid name "cell ranges", as it is a bit confusing. Those ranges are "subarray ranges", defined on a per dimension basis (i.e., they are 1D).